karapace/kafka_rest_apis/consumer_manager.py

```diff
@@ -160,30 +163,57 @@ def create_consumer(self, group_name: str, request_data: dict, content_type: str
             base_uri = urljoin(self.hostname, f"consumers/{group_name}/instances/{consumer_name}")
             KarapaceBase.r(content_type=content_type, body={"base_uri": base_uri, "instance_id": consumer_name})
 
-    def delete_consumer(self, internal_name: str, content_type: str):
+    async def create_kafka_consumer(self, fetch_min_bytes, group_name, internal_name, request_data):
+        while True:
+            try:
+                c = KafkaConsumer(
+                    bootstrap_servers=self.config["bootstrap_uri"],
+                    client_id=internal_name,
+                    security_protocol=self.config["security_protocol"],
+                    ssl_cafile=self.config["ssl_cafile"],
+                    ssl_certfile=self.config["ssl_certfile"],
+                    ssl_keyfile=self.config["ssl_keyfile"],
+                    group_id=group_name,
+                    fetch_min_bytes=fetch_min_bytes,
+                    request_timeout_ms=request_data["consumer.request.timeout.ms"],
+                    enable_auto_commit=request_data["auto.commit.enable"],
+                    auto_offset_reset=request_data["auto.offset.reset"]
+                )
+                return c
+            except:  # pylint: disable=bare-except
+                self.log.exception("Unable to create consumer, retrying")
+                await asyncio.sleep(1)
+
+    async def delete_consumer(self, internal_name: Tuple[str, str], content_type: str):
         self.log.info("Deleting consumer for %s", internal_name)
         self._assert_consumer_exists(internal_name, content_type)
-        with self.consumer_locks[internal_name]:
+        async with self.consumer_locks[internal_name]:
             try:
                 c = self.consumers.pop(internal_name)
                 c.consumer.close()
                 self.consumer_locks.pop(internal_name)
+            except:  # pylint: disable=bare-except
+                self.log.exception("Unable to properly dispose of consumer")
             finally:
                 empty_response()
 
     # OFFSETS
-    def commit_offsets(self, internal_name: str, content_type: str, request_data: dict, cluster_metadata: dict):
+    async def commit_offsets(
+        self, internal_name: Tuple[str, str], content_type: str, request_data: dict, cluster_metadata: dict
+    ):
         self.log.info("Committing offsets for %s", internal_name)
         self._assert_consumer_exists(internal_name, content_type)
         self._assert_has_key(request_data, "offsets", content_type)
         payload = {}
         for el in request_data["offsets"]:
+            for k in ["partition", "offset"]:
+                convert_to_int(el, k, content_type)
```

Comment:
Note: check if this is needed

Reply:
updated the aiven-core code, but the rest proxy still succeeded with the old code that passed strings instead of integers, which leads me to believe we should keep this